dapplet/src/index.ts

```diff
@@ -151,121 +145,214 @@ export default class TwitterFeature {
+      const prevSessions = await Core.sessions();
+      const prevSession = prevSessions.find((x) => x.authMethod === 'near/' + this._network);
+      const session = prevSession ?? (await Core.login({ authMethods: ['near/' + this._network] }));
+      const wallet = await session.wallet();
+      if (!nearAccountsFromCA.includes(wallet.accountId)) {
+        alert(
+          'You have connected accounts: ' + nearAccountsFromCA.join(', ') + '. Login with one of them to continue.',
+        );
+      } else {
+        const txHash = await this.tippingService.claimTokens(accountGId);
+        const explorerUrl =
+          this._network === NearNetworks.Mainnet ? 'https://explorer.near.org' : 'https://explorer.testnet.near.org';
+        alert(
+          `Claimed ${availableTokens} $NEAR with @tippingdapplet. ` + `Tx link: ${explorerUrl}/transactions/${txHash}`,
+        );
+      }
     } catch (e) {
       console.error(e);
     } finally {
       me.disabled = false;
       me.loading = false;
-      this.onProfileButtonClaimInit(profile, me);
+      this.executeInitWidgetFunctions();
     }
   };
 
   onProfileButtonLinkInit = async (profile, me) => {
-    const username = await this.getCurrentUserAsync();
+    const { username, websiteName } = await this.getCurrentUserAsync();
     const isMyProfile = profile.id.toLowerCase() === username?.toLowerCase();
-    const parsingNearAccount = this.parseNearId(profile.authorFullname, this._network);
+    const parsedNearAccount = this.parseNearId(profile.authorFullname, this._network);
 
     if (isMyProfile) {
-      const nearAccount = await this.identityService.getNearAccount('twitter/' + profile.id, true);
-      if (!nearAccount) {
-        me.label = 'Link';
-        me.tooltip = `Link ${parsingNearAccount ? parsingNearAccount + ' ' : ''}account with NEAR wallet`;
-      } else {
+      this._initWidgetFunctioins[[websiteName, username, 'link'].join('/')] = () =>
+        this.onProfileButtonLinkInit(profile, me);
+      const connectedAccounts = await Core.connectedAccounts.getConnectedAccounts(username, 'twitter');
+      const walletForAutoclaim = await this.tippingService.getWalletForAutoclaim(
+        username + '/' + websiteName.toLowerCase(),
+      );
+      const pendingRequestsIds = await Core.connectedAccounts.getPendingRequests();
+      let pendingRequest = null;
+      let madeRequestId = -1;
+      if (!walletForAutoclaim && pendingRequestsIds && pendingRequestsIds.length) {
+        const requests = await Promise.all(
+          pendingRequestsIds.map((pendingRequest) => Core.connectedAccounts.getVerificationRequest(pendingRequest)),
+        );
+        const gId = username + '/' + websiteName.toLowerCase();
+        requests.forEach((request, i) => {
+          if (request.firstAccount === gId || request.secondAccount === gId) {
+            pendingRequest = request;
+            madeRequestId = pendingRequestsIds[i];
+          }
+        });
+      }
```

Comment:
Can this logic be moved to a separate service to make the dapplet code cleaner?

Reply:
fixed